### PR TITLE
more js fixes

### PR DESCRIPTION
**Pull request by ricardoquesada**
fixes in how ref count is measured
opengl bindings converted to new API

### CODE REVIEW

File: cocos/base/CCRef.cpp [hunk 1]
```diff
@@ -111,7 +116,7 @@ void Ref::release()
     --_referenceCount;
 
 #if CC_ENABLE_SCRIPT_BINDING && CC_ENABLE_GC_FOR_NATIVE_OBJECTS
-    if (_scriptOwned && _referenceCount==1 && _rooted)
+    if (_scriptOwned && _rooted && _referenceCount==_referenceCountAtRootTime)
```
Comment (pandamicro) [inline cocos/base/CCRef.cpp:119]: Why is this necessary ? Sorry, I don't understand

Comment (ricardoquesada) [inline cocos/base/CCRef.cpp:119]: The thing is that I don't have to unroot the obj when refCount==1.
I have to unroot it when the refCount == sameRefCount when the object was rooted.

Comment (pandamicro) [inline cocos/base/CCRef.cpp:119]: But we don't know the retain release order, it may be retained by A then B then C, and released in order of B A C, I can't see why it's useful. For the rest of the PR, I have no problem

Comment (ricardoquesada) [inline cocos/base/CCRef.cpp:119]: You are right... We don't know the release order, but also we don't know if unrooting when rc==1 is the best solution since it can leak... in any case we need more testing.
